### ISSUE TITLE
Project References not working when projects do not follow a flat hierarchy

### ISSUE THREAD

**Issue by JulienTheron** [Microsoft/VSLinux#287]
Hi,

I'm using the official VS2017 15.7 just released yesterday. I discovered that project references do not work when the projects are not all in the same folder. Headers are not resolved when using the new "Public Project Include Directories" property, and linking fails.

Here are some repro steps:
1. Create a folder named "src". Create two subfolders, "apps" and "libs".
2. Create a Linux Console application "MyApp" under src/apps.
3. Create a Linux Library "MyLib" (either static or dynamic) under src/libs.
4. Place your .sln file under src.
5. Add the library as Reference to the Console application.
6. Add a class to the library and use it in the main.cpp file.
7. Add "./" to the Public Project Include Directories of the library project, then build.
8. main.cpp fails to build.

As a workaround for the build, add "../MyLib/" to the Additional Include Directories of the Console application project, then try to build again.
--> The build fails with `g++ : error : /home/julien/projects/MyApp/../../libs/MyLib/bin/x64/Debug/libMyLib.a: No such file or directory`
Unfortunately, there is no workaround apart from not using the project reference, which is needed to guarantee build order.

It seems the remote paths are not resolved correctly, as the apps and libs folders do not exist on the Linux host.

Thanks.

**Comment by yuehuang010** [comment]
Thanks, I am looking at it.

**Comment by levhaikin** [comment]
same thing for me...

**Comment by levhaikin** [comment]
related to #63

**Comment by yuehuang010** [comment]
Hello, here is a quick update.  I had to redesign the way P2P are handled for remote Linux.  Before the model was child project provide local path to parent project, then the parent project would translate it to remote path.  However, remote build takes a project centric model and assumes every project is leveled.  Thus the prior logic is wrong. 
With the up-coming 15.8 Preview 5, the new model will have the child project provide remote paths via a new metadata.  The parent project will use the remotepath without translation.  The theory is that only the current project could correctly know its own content and is the only one that could provide remote paths. 
Let me know if this address your problem.  Thanks.

**Comment by dt-karl-kuhn** [comment]
Hi @yuehuang010, I have just installed 15.8 Preview 5 and it had broken my build because of this issue. There is no clear documentation on how the remote paths should be set up if using a slightly more complex project/folder hierarchy.

e.g. My folder hierachy on linux should be 

```

-- RootFolder
  |-- SharedProjectSourcesFolder
  |-- SharedProjectBaseClassFolder
  |-- ProjectToBuildFolder
     |-- obj/../..
     |-- bin/../..

```

So I want to build everything in ProjectToBuildFolder which includes sources and headers in the shared folders. The output should sit under the ProjectToBuildFolder.

The project properties for settings where files are copied, where the build happens, and how other projects are includes are limited (Remote Build Root Directorty and Remote Build Project Directory).

In light of your changes, it would be really useful to have an example of how this would all map out, and where the project properties are used and where they vary, so that we can work out how to include items in other folders and also know where files will be copied to.

**Comment by JulienTheron** [comment]
Thanks @yuehuang010, the issue is fixed in 15.8 Preview 5!

**Comment by lukka** [comment]
Thanks for reporting it! Closing the issue.
